a part from the PR, I just tried similarity search by the example from docs:
```python
from ultralytics import solutions

app = solutions.SearchApp(
    # data = "path/to/img/directory" # Optional, build search engine with your own images
    device="cpu"  # configure the device for processing i.e "cpu" or "cuda"
)

app.run(debug=False)  # You can also use `debug=True` argument for testing
```
but it does not work to me, it went by error:
```text
Traceback (most recent call last):
  File "/home/laughing/codes/ultralytics/runs/1.py", line 877, in <module>
    test_similarity_search()
  File "/home/laughing/codes/ultralytics/runs/1.py", line 870, in test_similarity_search
    app = solutions.SearchApp(
          ^^^^^^^^^^^^^^^^^^^^
  File "/home/laughing/codes/ultralytics/ultralytics/solutions/similarity_search.py", line 207, in __init__
    from flask import Flask, render_template, request
  File "/home/laughing/anaconda3/lib/python3.11/site-packages/flask/__init__.py", line 5, in <module>
    from .app import Flask as Flask
  File "/home/laughing/anaconda3/lib/python3.11/site-packages/flask/app.py", line 30, in <module>
    from werkzeug.urls import url_quote
ImportError: cannot import name 'url_quote' from 'werkzeug.urls' (/home/laughing/anaconda3/lib/python3.11/site-packages/werkzeug/urls.py)
```
which I guess it's related to the version of `flask` package, probably we should pin a version for it?
FYI here's my test version.
![pic-250619-1825-03](https://github.com/user-attachments/assets/25705d43-d394-40ec-bca4-64ea6b27c13d)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -104,7 +104,7 @@ export = [
 solutions = [
     "shapely>=2.0.0",    # shapely for point and polygon data matching
     "streamlit>=1.29.0",    # for live inference on web browser, i.e `yolo streamlit-predict`
-    "flask",  # for similarity search solution
+    "flask>=3.0.1",  # for similarity search solution
 ]
 logging = [
     "wandb", # https://docs.ultralytics.com/integrations/weights-biases/

--- a/ultralytics/solutions/similarity_search.py
+++ b/ultralytics/solutions/similarity_search.py
@@ -206,7 +206,7 @@ class SearchApp:
             data (str, optional): Path to directory containing images to index and search.
             device (str, optional): Device to run inference on (e.g. 'cpu', 'cuda').
         """
-        check_requirements("flask")
+        check_requirements("flask>=3.0.1")
         from flask import Flask, render_template, request
 
         self.render_template = render_template

--- a/ultralytics/solutions/similarity_search.py
+++ b/ultralytics/solutions/similarity_search.py
@@ -70,9 +70,6 @@ class VisualAISearch(BaseSolution):
             safe_download(url=f"{ASSETS_URL}/images.zip", unzip=True, retry=3)
             self.data_dir = Path("images")
 
-        self.LOGGER.info(
-            f"Loading CLIP model '{self.model_name}' on {self.device}. This may download the model if not cached"
-        )
         self.model, self.preprocess = self.clip.load(self.model_name, device=self.device)
 
         self.index = None

